operators/pkg/custom-metrics-server/container_stats.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2022 Politecnico di Torino
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package custom_metrics_server
+
+import (
+	"context"
+
+	// runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1".
+	"github.com/go-logr/logr"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+// listContainerStats calls ListConteinerStats which sends a ListContainerStatsRequest to the server,
+// and parses the returned ListContainerStatsResponse.
+func listContainerStats(ctx context.Context, log logr.Logger, id *string) (*runtimeapi.ListContainerStatsResponse, error) {
+	filter := &runtimeapi.ContainerStatsFilter{}
+	if id != nil {
+		filter.Id = *id
+	}
+
+	runtimeClient := getRuntimeService(ctx, log.WithName("get-runtime-service"))
```

Comment:
Instead of using retrieving the runtime service, it would more idiomatic to make it the receiver of this function. BTW, this function seems a practically useless wrapper to me.

Reply:
Wrapper functions that build request filters based on parameters have been dropped. Now runtime api calls are made with a direct call to a given `runtimeClient`.